lib/github/ldap/user_search/active_directory.rb

```diff
@@ -0,0 +1,51 @@
+module GitHub
+  class Ldap
+    module UserSearch
+      class ActiveDirectory < Default
+
+        private
+
+        # Private - Overridden from base class to set the base to "", and use the
+        # Global Catalog to perform the user search.
+        def search(search_options)
+          Array(global_catalog_connection.search(search_options.merge(options)))
+        end
+
+        def global_catalog_connection
+          GlobalCatalog.connection(ldap)
+        end
+
+        # When doing a global search for a user's DN, set the search base to blank
+        def options
+          super.merge(base: "")
+        end
+      end
+
+      class GlobalCatalog < Net::LDAP
+        STANDARD_GC_PORT = 3268
+        LDAPS_GC_PORT = 3269
```

Comment:
Surprised we'd inherit from `Net::LDAP` rather than just wrap it and expose a minimal API.